packages/components/src/styles/components/form/layout.scss

```diff
@@ -43,12 +43,10 @@
 
 .hds-form__field-group {
   // When there are columns, stack them vertically on small screens
-  &.hds-layout-flex--direction-row {
```

Comment:
I added this to increase the specificity so the styling wouldn't get overridden. When you deleted this it broke the responsive styling.